src/common/lib/util/defaults.ts

```diff
@@ -295,6 +295,17 @@ export function normaliseOptions(options: DeprecatedClientOptions): NormalisedCl
     }
   }
 
+  let internetAvailabilityParams = null;
+  if (options.internetAvailabilityUrl) {
+    let [uri, qs] = options.internetAvailabilityUrl.split('?');
+    const params = qs ? Utils.parseQueryString(qs) : {};
+    if (uri.indexOf('://') === -1) {
+      uri = (options.tls ? 'https://' : 'http://') + uri;
+    }
+    options.internetAvailabilityUrl = uri;
```

Comment:
What's the use case for mutating this URL? Should it not just be opaque? What if the user wants to use TLS with Ably, but for whatever reason, their internet availability URL is non-TLS?

Reply:
I've changed it a bit now so you can use plain HTTP if you pass in a scheme but otherwise it defaults to TLS.

As for mutating the URL, it's just easier to normalise it once on init I guess.